cypress/e2e/right-panel/right-panel.spec.ts

```diff
@@ -65,74 +61,89 @@ describe("RightPanel", () => {
         it("should handle clicking add widgets", () => {
             viewRoomSummaryByName(ROOM_NAME);
 
-            cy.get(".mx_RoomSummaryCard_appsGroup .mx_AccessibleButton").click();
+            cy.findByRole("button", { name: "Add widgets, bridges & bots" }).click();
             cy.get(".mx_IntegrationManager").should("have.length", 1);
         });
 
         it("should handle viewing export chat", () => {
             viewRoomSummaryByName(ROOM_NAME);
 
-            cy.get(".mx_RoomSummaryCard_icon_export").click();
+            cy.findByRole("button", { name: "Export chat" }).click();
             cy.get(".mx_ExportDialog").should("have.length", 1);
         });
 
         it("should handle viewing share room", () => {
             viewRoomSummaryByName(ROOM_NAME);
 
-            cy.get(".mx_RoomSummaryCard_icon_share").click();
+            cy.findByRole("button", { name: "Share room" }).click();
             cy.get(".mx_ShareDialog").should("have.length", 1);
         });
 
         it("should handle viewing room settings", () => {
             viewRoomSummaryByName(ROOM_NAME);
 
-            cy.get(".mx_RoomSummaryCard_icon_settings").click();
+            cy.findByRole("button", { name: "Room settings" }).click();
             cy.get(".mx_RoomSettingsDialog").should("have.length", 1);
-            cy.get(".mx_Dialog_title").should("contain", ROOM_NAME);
+            cy.get(".mx_Dialog_title").within(() => {
+                cy.findByText("Room Settings - " + ROOM_NAME).should("exist");
+            });
         });
 
         it("should handle viewing files", () => {
             viewRoomSummaryByName(ROOM_NAME);
 
-            cy.get(".mx_RoomSummaryCard_icon_files").click();
+            cy.findByRole("button", { name: "Files" }).click();
             cy.get(".mx_FilePanel").should("have.length", 1);
             cy.get(".mx_FilePanel_empty").should("have.length", 1);
 
-            goBack();
+            cy.findByRole("button", { name: "Room information" }).click();
             checkRoomSummaryCard(ROOM_NAME);
         });
 
         it("should handle viewing room member", () => {
             viewRoomSummaryByName(ROOM_NAME);
 
-            cy.get(".mx_RoomSummaryCard_icon_people").click();
+            // \d represents the number of the room members inside mx_BaseCard_Button_sublabel
+            cy.findByRole("button", { name: /People \d/ }).click();
```

Comment:
Yes, I agree it's better to just assert one thing at a time rather than effectively checking the people number is correct too. The code comment is helpful, thanks!